inference-engine/ie_bridges/java/cpp/input_info.cpp

```diff
@@ -0,0 +1,126 @@
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+}
+
+JNIEXPORT void JNICALL Java_org_intel_openvino_InputInfo_SetLayout(JNIEnv *env, jobject obj, jlong addr, jint layout)
+{
+    static const char method_name[] = "SetLayout";
+    try
+    {
+        InputInfo *input_info = (InputInfo*)addr;
+        auto it = layout_map.find(layout_e(layout));
+
+        if (it == layout_map.end())
+            throw std::runtime_error("No such layout value!");
+
+        input_info->setLayout(it->second);
+
+    } catch (const std::exception &e){
+        throwJavaException(env, &e, method_name);
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+}
+
+JNIEXPORT jobject JNICALL Java_org_intel_openvino_InputInfo_getLayout(JNIEnv *env, jobject, jlong addr){
+    static const char method_name[] = "getLayout";
+    try
+    {
+        InputInfo *input_info = (InputInfo*)addr;
+        Layout layout = input_info->getLayout();
+
+        auto layout_res = find_by_value(layout_map, layout);
+
+        jclass layout_class = env->FindClass("org/intel/openvino/Layout");
+        jmethodID valueOf_method_id = env->GetStaticMethodID(layout_class, "valueOf", "(I)Lorg/intel/openvino/Layout;");
+        jobject layout_object = env->CallStaticObjectMethod(layout_class, valueOf_method_id, layout_res);
+
+        return layout_object;
+
+    } catch (const std::exception &e){
+        throwJavaException(env, &e, method_name);
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+}
+
+JNIEXPORT void JNICALL Java_org_intel_openvino_InputInfo_SetPrecision(JNIEnv *env, jobject obj, jlong addr, jint precision)
+{
+    static const char method_name[] = "SetPrecision";
+    try
+    {
+        InputInfo *input_info = (InputInfo*)addr;
+        auto it = precision_map.find(precision_e(precision));
+
+        if (it == precision_map.end())
+            throw std::runtime_error("No such precision value!");
+
+        input_info->setPrecision(it->second);
+        
+    } catch (const std::exception &e)
+    {
+        throwJavaException(env, &e, method_name);
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+}
+
+JNIEXPORT jobject JNICALL Java_org_intel_openvino_InputInfo_getPrecision(JNIEnv *env, jobject, jlong addr){
+    static const char method_name[] = "getPrecision";
+    try
+    {
+        InputInfo *input_info = (InputInfo*)addr;
+        Precision precision = input_info->getPrecision();
+
+        auto precision_res = find_by_value(precision_map, precision);
+
+        jclass precision_class = env->FindClass("org/intel/openvino/Precision");
+        jmethodID valueOf_method_id = env->GetStaticMethodID(precision_class, "valueOf", "(I)Lorg/intel/openvino/Precision;");
+        jobject precision_object = env->CallStaticObjectMethod(precision_class, valueOf_method_id, precision_res);
+
+        return precision_object;
```

Comment:
Can return also `jint` similar to `Java_org_intel_openvino_TensorDesc_getLayout`?